src/transform/FooterMenu.js

```diff
@@ -0,0 +1,215 @@
+import './FooterMenu.css'
+
+/**
+ * @typedef {function} FooterMenuItemPayloadExtractor
+ * @param {!Document} document
+ * @return {!string[]} Important - should return empty array if no payload strings.
+ */
+
+/**
+  * @typedef {function} FooterMenuItemClickCallback
+  * @param  {!string[]} payload Important - should return empty array if no payload strings.
+  * @return {void}
+  */
+
+/**
+ * Extracts array of no-html page issues strings from document.
+ * @type {FooterMenuItemPayloadExtractor}
+ */
+const pageIssuesStringsArray = document => {
+  const tables = document.querySelectorAll(
+    'div#content_block_0 table.ambox:not(.ambox-multiple_issues):not(.ambox-notice)'
+  )
+  // Get the tables into a fragment so we can remove some elements without triggering a layout
+  const fragment = document.createDocumentFragment()
+  for (let i = 0; i < tables.length; i++) {
+    fragment.appendChild(tables[i].cloneNode(true))
+  }
+  // Remove some element so their text doesn't appear when we use "innerText"
+  Array.from(fragment.querySelectorAll('.hide-when-compact, .collapsed')).forEach(el => el.remove())
+  // Get the innerText
+  return Array.from(fragment.querySelectorAll('td[class$=mbox-text]')).map(el => el.innerText)
+}
+
+/**
+ * Extracts array of disambiguation page urls from document.
+ * @type {FooterMenuItemPayloadExtractor}
+ */
+const disambiguationTitlesArray = document => Array.from(
+  document.querySelectorAll(
+    'div#content_block_0 div.hatnote a[href]:not([href=""]):not([redlink="1"])'
+  )
+).map(el => el.href)
+
+/**
+ * Type representing kinds of menu items.
+ * @type {Object}
+ */
+const MenuItemType = {
+  languages: 1,
+  lastEdited: 2,
+  pageIssues: 3,
+  disambiguation: 4,
+  coordinate: 5
+}
+
+/**
+ * Menu item model.
+ */
+class WMFMenuItem {
+  /**
+   * WMFMenuItem constructor.
+   * @param  {!string} title
+   * @param  {?string} subtitle
+   * @param  {!MenuItemType} itemType
+   * @param  {FooterMenuItemClickCallback} clickHandler
+   * @return {void}
+   */
+  constructor(title, subtitle, itemType, clickHandler) {
+    this.title = title
+    this.subtitle = subtitle
+    this.itemType = itemType
+    this.clickHandler = clickHandler
+    this.payload = []
+  }
+
+  /**
+   * Returns icon CSS class for this menu item based on its type.
+   * @return {!string}
+   */
+  iconClass() {
+    switch (this.itemType) {
```

Comment:
I chewed on this for a bit... thinking I'd prefer to keep `MenuItemType` an enum. Also was curious as to the reason for opposition to switch... Thoughts?